pkg/kubectl/cmd/templates/command_groups.go

```diff
@@ -35,6 +35,20 @@ func (g CommandGroups) Add(c *cobra.Command) {
 	}
 }
 
+// PruneCommands removes commands that are set to nil. This allows commands to disable
+// themselves by returning a nil cobra.Command.
+func (g CommandGroups) PruneCommands() {
+	for i, group := range g {
+		gc := make([]*cobra.Command, 0, len(group.Commands))
+		for _, c := range group.Commands {
+			if c != nil {
+				gc = append(gc, c)
+			}
+		}
+		g[i].Commands = gc
+	}
+}
```

Comment:
Yup, that's what I intended here, but what you describe could definitely be a better user experience. I'll try to figure out something for this in #46151